cis194/week6/zhansongl/Fibonacci.hs

```diff
@@ -25,13 +27,13 @@ instance Show a => Show (Stream a) where
   show = show . take 20 . streamToList
 
 streamRepeat :: a -> Stream a
-streamRepeat a = Cons a . streamRepeat $ a
+streamRepeat = Cons <*> streamRepeat
 
 streamMap :: (a -> b) -> Stream a -> Stream b
 streamMap f (Cons h t) = Cons (f h) (streamMap f t)
 
 streamFromSeed :: (a -> a) -> a -> Stream a
-streamFromSeed f s = Cons s (streamFromSeed f (f s))
+streamFromSeed f = Cons <*> (streamFromSeed f . f)
```

Comment:
```suggestion
streamFromSeed f = Cons <*> streamFromSeed f . f
```
A lot operators are designed to have the precedence just right, so that we don't have to put brackets around function compositions before and after. `<*>` is one of them.